compiler/util/clangUtil.cpp

```diff
@@ -1215,52 +1205,88 @@ bool setAlreadyConvertedExtern(ModuleSymbol* module, const char* name)
 void runClang(const char* just_parse_filename) {
   static bool is_installed_fatal_error_handler = false;
 
-  /* TODO -- note that clang/examples/clang-interpreter/main.cpp
-             includes an example for getting the executable path,
-             so that we could automatically set CHPL_HOME. */
-  std::string home(CHPL_HOME);
-  std::string compileline = home + "/util/config/compileline";
-  compileline += " COMP_GEN_WARN="; compileline += istr(ccwarnings);
-  compileline += " COMP_GEN_DEBUG="; compileline += istr(debugCCode);
-  compileline += " COMP_GEN_OPT="; compileline += istr(optimizeCCode);
-  compileline += " COMP_GEN_SPECIALIZE="; compileline += istr(specializeCCode);
-  compileline += " COMP_GEN_FLOAT_OPT="; compileline += istr(ffloatOpt);
-
-  std::string readargsfrom;
-
-  if( !llvmCodegen && just_parse_filename ) {
-    // We're handling an extern block and not using the LLVM backend.
-    // Don't ask for any compiler-specific C flags.
-    readargsfrom = compileline + " --llvm"
-                                 " --clang"
-                                 " --clang-sysroot-arguments"
-                                 " --includes-and-defines";
-  } else {
-    // We're parsing extern blocks AND any parts of the runtime
-    // in order to prepare for an --llvm compilation.
-    // Use compiler-specific flags for clang-included.
-    readargsfrom = compileline + " --llvm"
-                                 " --clang"
-                                 " --clang-sysroot-arguments"
-                                 " --cflags"
-                                 " --includes-and-defines";
-  }
+  const char* clang_warn[] = {"-Wall", "-Werror", "-Wpointer-arith",
```

Comment:
Seems a little weird that we're duplicating flags here instead of relying on the existing `make/compiler/Makefile.*` infrastructure, but I don't really understand this code to know if there's a good reason for this.

Reply:
The whole point is to avoid shelling out to other things. The main job of the previous Perl compileline script was actually to run some Make targets so that we could use the make/compiler/Makefile.* makefiles (among other things). However that design doesn't make so much sense because there's only one compiler we're working with here - clang - and even mostly only one version - the version we bundle.